drivers/include/digit7seg.h

```diff
@@ -0,0 +1,202 @@
+
+/**
+ * @brief   Initialize the given DIGIT7SEG
+ *
+ * @param[out] dev      DIGIT7SEG device descriptor to initialize
+ * @param[in]  params   Device parameters to use
+ *
+ * @return  0 on success or negative error code, see #digit7seg_error_codes
+ */
+int digit7seg_init(digit7seg_t *dev, const digit7seg_params_t *params);
+
+/**
+ * @brief   Set the value for every digit
+ *
+ * @param[in] dev       Initialized device descriptor of DIGIT7SEG device
+ * @param[in] value     the value as an uint32_t
+ *
+ */
+void digit7seg_set_all_value(digit7seg_t *dev, uint32_t value);
+
+/**
+ * @brief   Set the value for one digit
+ *
+ * @param[in] dev       Initialized device descriptor of DIGIT7SEG device
+ * @param[in] index     the digit number between [0, DIGIT7SEG_MAX_DIGITS[
+ * @param[in] value     the value as an uint8_t, a is equal to the first bit
+ *                       b the second.... dp the 8th bit.
+ *
+ * @return              0 on success
+ * @return              -1 when an invalid index is passed
+ */
+int digit7seg_set_value(digit7seg_t *dev, int index, uint8_t value);
+
+#if MODULE_DIGIT7SEG_INT || DOXYGEN
+/**
+ * @brief   Set the value for every digit from an int
+ *
+ * @param[in] dev       Initialized device descriptor of DIGIT7SEG device
+ * @param[in] value     the value between [0, 9999]
+ *
+ */
+int digit7seg_set_int_all_value(digit7seg_t *dev, uint16_t value);
+
+/**
+ * @brief   Set the value for one digit from an int
+ *
+ * @param[in] dev       Initialized device descriptor of DIGIT7SEG device
+ * @param[in] index     the digit number between [0, DIGIT7SEG_MAX_DIGITS[
+ * @param[in] value     the value between [0, 9]
+ *
+ */
+int digit7seg_set_int_value(digit7seg_t *dev, int index, uint8_t value);
+#endif
+
+#if MODULE_DIGIT7SEG_FLOAT || DOXYGEN
+/**
+ * @brief   Set a float value dispatched between every display
+ *
+ * @param[in] dev       Initialized device descriptor of DIGIT7SEG device
+ * @param[in] value     A float value between [0, 999,9]
+ * @param[in] precision The multiplier factor ex: value = 234,15, precision = 10
+ *                       will be displayed as 234,1 but with
+ *                       precision = 100 it will be 34,15
+ */
+int digit7seg_set_float_value(digit7seg_t *dev, float value, int precision);
+#endif
+
+/**
+ * @brief   Start an periodic timer event to shift between each 7seg
+ *
+ * @param[in] dev       Initialized device descriptor of DIGIT7SEG device
+ *
+ * @return              0 on success
+ * @return              -1 if the timer was impossible to start
+ */
+int digit7seg_poweron(digit7seg_t *dev);
+
+/**
+ * @brief   Stop an periodic timer event to shift between each 7seg
+ *
+ * @param[in] dev       Initialized device descriptor of DIGIT7SEG device
+ */
+void digit7seg_poweroff(digit7seg_t *dev);
+
+#if MODULE_DIGIT7SEG_INT || MODULE_DIGIT7SEG_FLOAT || DOXYGEN
+/**
+ * @brief   Display bitfield for numbers between [0, 9]
+ */
+static const uint8_t digit7seg_bitfield[] = {
```

Comment:
Should this really be in the header?

Every compilation unit that actually makes use of this array will end up adding its own copy. If it would be added to a C file, we don't get duplicates.